### PR TITLE
upgrade dependencies, support for actix-web v4

### PR DESCRIPTION
**Pull request by sbruton**
this PR upgrades the dependencies to their latest stable releases

notably, it now supports the current stable actix-web v4 middleware changes

this does, however, break support for previous actix-web v3

### CODE REVIEW

**Comment by vbrandl** [comment]
Thanks for your contribution, I'll look into it, soon and I guess, I'll just publish a `4.0.0` release to match the `actix-web` major version.

Could you try fixing the tests? It should only require adding `use std::io::Read;` in the test module.

**Comment by sbruton** [comment]
yep, on it

**Comment by sbruton** [comment]
tests fixed

**Comment by sbruton** [comment]
i should update the readme as well, to show the actix-web v4 format, i'll do that now

**Comment by sbruton** [comment]
there ya go, all done for now unless you see need for further improvements

File: README.md [hunk 1]
```diff
@@ -18,23 +18,24 @@ the request using `RequestId`.
...
+async fn show_request_id(id: RequestId) -> impl Responder {
+    format!("{}", id.as_str())
 }
 
-#[actix_rt::main]
-async fn main() {
-    HttpServer::new(move || {
+#[actix_web::main] // or #[tokio::main]
+async fn main() -> std::io::Result<()> {
+    let http_server = HttpServer::new(|| {
```
Comment (vbrandl) [inline README.md]: Shouldn't the result of `...run().await` be returned from `main`? I think the `let http_server =` can be removed, or `http_server` should be returned in the end.

**Comment by vbrandl** [comment]
Could you also add the following section to the `README.md`:

```
# Supported `actix-web` Versions

| crate version | `actix-web` version |
| ---           | ---                 |
| `0.1.0`       | `v2`                |
| `0.2.0`       | `v3`                |
| `4.0.0`       | `v4`                |
```

**Comment by sbruton** [comment]
done, also fixed the docs badge

**Comment by vbrandl** [comment]
LGTM, I'll merge now and publish a new release to crates.io

**Comment by vbrandl** [comment]
Well... The release will have to wait until GitHub Actions are working properly again...

**Comment by vbrandl** [comment]
`4.0.0` has been published on crates.io: https://crates.io/crates/actix-request-identifier 

You can now use the crate with actix-web v4

**Pull request merged**